### PR TITLE
Feat/registration

### PR DESCRIPTION
**Pull request by DavidKorystov**

### CODE REVIEW

File: src/containers/Registration.tsx [hunk 1]
```diff
@@ -204,8 +204,8 @@ const Registration = () => {
     <>
       <ModalUniversal
         button={registrationButtonName}
-        title={registrationTitle}
-        content={registerForm}
+        title={currentUserId ? "close me" : registrationTitle}
```
Comment (VitaliiDenysiuk) [inline src/containers/Registration.tsx:206]: use i18n.
do not write string without i18n

File: src/redux/reducers/index.ts [hunk 2]
```diff
@@ -2,11 +2,9 @@
 import { combineReducers } from "redux";
 
 import tweetsSlice from "./tweets";
-import { userSlice } from "./user";
+import userSlice from "./user";
```
Comment (VitaliiDenysiuk) [inline src/redux/reducers/index.ts]: import tweetsReducer from "./tweets";
import userReducer from "./user";

at that place we have reducers not slice!

Comment (VitaliiDenysiuk) [inline src/redux/reducers/index.ts]: import tweetsReducer from "./tweets";
import userReducer from "./user";

at that place we have reducers not slice!

File: src/redux/reducers/index.ts [hunk 3]
```diff
@@ -2,11 +2,9 @@
...
 
 export const rootReducer = combineReducers({
-  tweets: tweetsSlice.reducer,
-  user: userSlice.reducer,
+  tweets: tweetsSlice,
+  user: userSlice,
```
Comment (VitaliiDenysiuk) [inline src/redux/reducers/index.ts]:
```suggestion
  tweets: tweetsReducer,
  user: userReducer,
```

at that place we have reducers not slice!

File: src/redux/reducers/index.ts (hunk 2 above)
Comment (VitaliiDenysiuk) [inline src/redux/reducers/index.ts]: import tweetsReducer from "./tweets";
import userReducer from "./user";

at that place we have reducers not slice!

File: src/redux/reducers/user.ts [hunk 4]
```diff
@@ -31,26 +32,75 @@ const userInitialState: UserStore = {
...
+          local: {
+            accessToken: payload.accessToken,
+            refreshToken: payload.refreshToken,
+          },
+        },
+      };
+    });
+    builder.addCase(registerUser.rejected, (store) => {
+      store.isLoading = false;
+      store.error = "Failed to fetch";
```
Comment (VitaliiDenysiuk) [inline src/redux/reducers/user.ts]: add more details for error

File: src/services/api.ts [hunk 5]
```diff
@@ -111,9 +96,9 @@ export const loginUser = async (
...
     const res = await instance.post("/api/auth/login", {
-      username: username,
-      password: password,
-      email: email,
+      username,
+      password,
+      email,
     });
     const { accessToken, refreshToken } = res.data;
     localStorage.setItem("accessToken", accessToken);
```
Comment (VitaliiDenysiuk) [inline src/services/api.ts:104]: remove that 
```
localStorage.setItem("accessToken", accessToken);
```
. Now we use redux
and function loginUserAsLeleka1 change: delete  localStorage usage

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/containers/Registration.tsx (hunk 1 above)
Comment (DavidKorystov) [reply src/containers/Registration.tsx:206]: This is a temporary change in the display of the modal window. So there's no sense.

File: src/services/api.ts (hunk 5 above)
Comment (alexandrtovmach) [reply src/services/api.ts:104]: this functions is only for test reasons and can be completelly removed

File: src/types/index.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
 export interface MongoArtifacts {
-  _id: string;
+  _id?: string;
```
Comment (alexandrtovmach) [inline src/types/index.ts]:
```suggestion
  _id: string;
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/services/api.ts (hunk 5 above)
Comment (DavidKorystov) [reply src/services/api.ts:104]: Thanks, I was just about to add)

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**